include/eve/detail/skeleton.hpp

```diff
@@ -131,6 +135,21 @@ namespace eve::detail
     }
   }
 
+  // AGGREGATE Helper for fusing tuple
+  template<typename... T> struct combiner
+  {
+    using type = kumi::tuple<typename T::template rescale<typename T::cardinal_type::combined_type>...>;
+  };
+
+  template<typename T>
+  EVE_FORCEINLINE auto tuple_combiner(T const& lo, T const& hi)
+  {
+    using out_t = kumi::apply_traits_t<combiner,T>;
+    return kumi::map_index([](auto i, auto const& l, auto const& h) { return kumi::element_t<i,out_t>{l,h}; }
+                          , lo, hi
+                          );
+  }
```

Comment:
No. We have to aggregate functions that return a tuple of wide.
To do so, we slice the inputs, computes the resulting tuples and combine their subparts.
We can use eve::combine here for dependencies reasons so we just use the combining wide constructor, hence the beed of the subtype.

Reply:
so, it's aggregating all components in a tuple? can I have a comment

Reply:
There is two case of aggregation : aggregation over multiple register and AVX aggregation for integer wide than can be sliced in two because AVX don't have much operation on integers.

We store aggregates as an array of K regular native register, K being 2,4,8 or w/e. So if we end up in the case of two actual aggregate, we slice it up over those K slices as it is faster to compile. In the other case, we do the recursive slice and call

For the tuple return, a bunch of functions like sincos, rempio2, airy etc return a tuple of wide. Now, the real question is should it be a wide of tuple but it is out of scope for this PR.

So the previous version had special apply_over variant like apply_over2, apply_over3 that handled apply_over over function that returned a tuple so that aggregate worked correctly. The tuple_combiner is a streamlined and more compact way of doing it that also happen to work whatever the size of the output tuple.

If you think tuple of wide as return type should go, then maybe we cna devise that afterward but I needed the existing code to work atm.